protected/humhub/modules/admin/controllers/ApprovalController.php

```diff
@@ -77,28 +80,59 @@ public function beforeAction($action)
         return parent::beforeAction($action);
     }
 
+    /**
+     * @return string
+     * @throws \Throwable
+     * @throws \yii\base\InvalidConfigException
+     */
     public function actionIndex()
     {
         $searchModel = new UserApprovalSearch();
         $dataProvider = $searchModel->search(Yii::$app->request->queryParams);
 
+        // Get available profile fields for screen options
+        $availableProfileFields = ProfileField::find()
+            ->joinWith('category')
+            ->orderBy([
+                'profile_field_category.sort_order' => SORT_ASC,
+                'profile_field.sort_order'=>SORT_ASC
+            ])
+            ->where(['profile_field.show_at_registration' => true])
+            ->andWhere(['not', ['profile_field.internal_name' => ['firstname', 'lastname']]])
+            ->all();
+
+        // Get or set screen options
+        $userSettings = Yii::$app->settings->user(Yii::$app->user->identity);
+        $screenProfileFieldsId = $userSettings->getSerialized(self::USER_SETTINGS_SCREEN_KEY, []);
+        if (Yii::$app->request->post('screenProfileFieldsId')) {
+            $screenProfileFieldsId = Yii::$app->request->post('screenProfileFieldsId');
+            $userSettings->setSerialized(self::USER_SETTINGS_SCREEN_KEY, $screenProfileFieldsId);
+        }
+        $profileFieldsColumns = !$screenProfileFieldsId ? [] : ProfileField::find()
+            ->where(['id' => $screenProfileFieldsId])
+            ->indexBy('id')
+            ->all();
+
         return $this->render('index', [
             'dataProvider' => $dataProvider,
-            'searchModel' => $searchModel
+            'searchModel' => $searchModel,
+            'availableProfileFields' => $availableProfileFields,
+            'profileFieldsColumns' => $profileFieldsColumns,
         ]);
     }
 
     /**
      * @param $id
-     * @return string
-     * @throws HttpException
+     * @return string|\yii\console\Response|\yii\web\Response
      * @throws \Throwable
+     * @throws \yii\base\InvalidConfigException
      */
     public function actionApprove($id)
     {
         $model = new ApproveUserForm($id);
         $model->setApprovalDefaults();
         if($model->load(Yii::$app->request->post()) && $model->approve()) {
+            $this->view->success(Yii::t('AdminModule.controllers_ApprovalController', 'The user has been approved and the email has been sent'));
```

Comment:
Please use `AdminModule.user` - I'll move all existing translation strings `AdminModule.controllers_ApprovalController` to `AdminModule.user` after the PR merge.

Reply:
New text: `The registration was approved and the user was notified by email. `